sdk/go/x/auto/local_workspace.go

```diff
@@ -244,6 +245,37 @@ func (l *LocalWorkspace) RefreshConfig(ctx context.Context, fqsn string) (Config
 	return cfg, nil
 }
 
+// GetEnvVars returns the environment values scoped to the current workspace.
+func (l *LocalWorkspace) GetEnvVars() ([]string, error) {
+	if l.envvars == nil {
+		return nil, errors.New("unable to get environment values")
+	}
+	return l.envvars, nil
+}
+
+// SetEnvVars sets the specified map of environment values scoped to the current workspace.
```

Comment:
The comment here mentions a map, but the function parameter is an array. I wonder if accepting a map would actually be a good idea? If we use a map we can offer an `Unsetenv` API which would be consistent with the os package: https://golang.org/pkg/os/#Unsetenv

A map might also be easier for users to manipulate as it doesn't require parsing out key-value pairs from the array.

Reply:
A note on the doc comments. It might be worth mentioning that these values will be passed to all `Workspace` and `Stack` level commands like `Up`, `ListStacks`, `SetConfig` etc.

Reply:
>The comment here mentions a map, but the function parameter is an array. 
>The second comment refers to using a map[string]string as the underlying data structure to store env vars as well.

Originally `map[string]string` was used, hence the left over comments :).

We switched to a `[]string` part way through since this is the type used in the `cmd.Env` and what the issue originally called out for the API, but happy to switch it to a map.

> If we use a map we can offer an Unsetenv API which would be consistent with the os package: https://golang.org/pkg/os/#Unsetenv

Fair point. Though I wonder how often it will be used in these scenarios. Thoughts?

> A note on the doc comments. It might be worth mentioning that these values will be passed to all Workspace and Stack level commands like Up, ListStacks, SetConfig etc.

👍

Reply:
Ah, Indeed I did put that in the original issue. Sounds like you had the right intuition from the get-go, but I needed a minute to think about it :) 

> Fair point. Though I wonder how often it will be used in these scenarios. Thoughts?

I think that if we want to allow workspaces to be stateful and reusable, mutating the environment variables in some way would probably be helpful. A single Unsetenv method is probably enough to satisfy this, but I'll leave it up to you.